File: src/main/java/com/todolist/task/service/adapter/mysql/models/Auditable.java
package com.todolist.task.service.adapter.mysql.models;

import jakarta.persistence.Column;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@EntityListeners(AuditingEntityListener.class)
@MappedSuperclass
@Getter
@Setter
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
public abstract class Auditable {

  String createdBy;

  @CreatedDate
  @Column(nullable = false, updatable = false)
  Instant createdTime;
  @LastModifiedBy
  String updatedBy;

  @LastModifiedDate
  Instant updatedTime;
}

File: src/main/kotlin/com/todolist/task/service/adapter/mysql/models/Auditable.kt
package com.todolist.task.service.adapter.mysql.models

import jakarta.persistence.Column
import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.Instant

@EntityListeners(AuditingEntityListener::class)
@MappedSuperclass
abstract class Auditable {
    @CreatedDate
    @Column(nullable = false, updatable = false)
    var createdTime: Instant = Instant.now()

    @LastModifiedDate
    @Column(nullable = false)
    var updatedTime: Instant = Instant.now()

    var createdBy: String? = null

    var updatedBy: String? = null
}
